### DISCUSSION TITLE
Recoded video has wrong orientation

### DISCUSSION THREAD

**Discussion by Str1atum** [immich-app/immich#10634] [Q&A]
With 1.106.4 if have an issue with the recoded videos being the wrong orientation.
It affects HEVC encoded videos in 1080p or 4k from iPhone 13 and 15, all HDR
Videos in portrait orientation are rotated to landscape. Videos in landscape are rotated 180°.
The videos do not just play that way in the browser but also when I download the recoded videos with Chrome to my local disk.

recode config is:

"ffmpeg": {
"crf": 28,
"threads": 0,
"preset": "medium",
"targetVideoCodec": "hevc",
"acceptedVideoCodecs": [
"hevc"
],
"targetAudioCodec": "aac",
"acceptedAudioCodecs": [
"libopus",
"mp3",
"aac"
],
"targetResolution": "720",
"maxBitrate": "2500",
"bframes": -1,
"refs": 0,
"gopSize": 0,
"npl": 0,
"temporalAQ": false,
"cqMode": "auto",
"twoPass": true,
"preferredHwDevice": "auto",
"transcode": "required",
"tonemap": "hable",
"accel": "qsv",
"accelDecode": true

**Comment by alextran1502** [comment]
I have iPhone 15 as well, let me try with your settings to see if I can reproduce

**Comment by Str1atum** [comment]
Make sure you enable HDR, it seems to be related to that feature.

**Comment by thorin29** [comment]
I'm on a Pixel 5, and my family members are on a Pixel 7 and a Pixel 7 Pro. Not sure what information you need but any newer video that is in landscape orientation is now inverted and plays upside down in the app. It was like this before the most recent update to 1.107.0 and still persists now. This is only in the Android app.

**Reply by alextran1502** [reply]
Alright, let me get a pixel test phone and troubleshoot this

**Reply by thorin29** [reply]
Hey Alex, any updates? If you need more info I can provide it. Thanks!

**Reply by apeman76** [reply]
> Hey Alex, any updates? If you need more info I can provide it. Thanks!

Fixed with https://github.com/immich-app/immich/pull/10905
1.108 iirc

**Reply by thorin29** [reply]
I must be missing something if the issue is resolved, my server is on v1.09.2 and my android app is up to date. The orientation is only wrong in the android app. The thumbnail looks correct, then as soon as it starts playing the video flips upside down.

![Screenshot_20240719-215542.png](https://github.com/user-attachments/assets/ea30281a-3aab-48db-b533-7ec03e7a5738)

**Reply by Str1atum** [reply]
You need to reencode the videos

**Reply by thorin29** [reply]
> You need to reencode the videos

What do I need to do to reencode? I did this: administration > jobs > transcode videos > all. And I still have the same issue.

Thanks for the help!

**Comment by apeman76** [comment]
I have the same issue, landscape videos are oriented to horizontal. They play that way on all devices, even on web. I have this happening on videos from i.e. snapchat saved to gallery aswell, I can submit original video file and recoded video file through discord if useful

Cause?
https://stackoverflow.com/questions/69386275/mp4-and-rotation-remove-flags-but-set-rotation/69396105#69396105

PR: https://github.com/immich-app/immich/pull/10905
https://discord.com/channels/979116623879368755/994044917355663450/1258912309829566494
